Detectors/Calibration/SpacePoints/src/TrackResiduals.cxx

```diff
@@ -0,0 +1,1379 @@
+      return (ix < param::NPadRows) ? ix : param::NPadRows - 1;
+    } else {
+      // pad size is uniform in OROC1
+      ix = (x - (param::RowX[param::NRowIROC] - 0.5 * param::RowDX[param::NRowIROC])) / param::RowDX[param::NRowIROC] + param::NRowIROC;
+      if (ix < param::NRowIROC) {
+        // we are in between IROC and OROC1
+        if (x > 0.5 * (param::RowX[param::NRowIROC - 1] + param::RowX[param::NRowIROC])) {
+          ix = param::NRowIROC; // 1st OROC1 row
+        } else {
+          ix = param::NRowIROC - 1;
+        }
+      }
+      return ix;
+    }
+  }
+}
+
+void TrackResiduals::setKernelType(int type, float bwX, float bwP, float bwZ, float scX, float scP, float scZ)
+{
+  // set kernel type and widths in terms of binning in x, y/x, z/x and define aux variables
+  mKernelType = type;
+
+  mKernelScaleEdge[param::VoxX] = scX;
+  mKernelScaleEdge[param::VoxF] = scP;
+  mKernelScaleEdge[param::VoxZ] = scZ;
+
+  mKernelWInv[param::VoxX] = (bwX > 0) ? 1. / bwX : 1.;
+  mKernelWInv[param::VoxF] = (bwP > 0) ? 1. / bwP : 1.;
+  mKernelWInv[param::VoxZ] = (bwZ > 0) ? 1. / bwZ : 1.;
+
+  if (mKernelType == param::EpanechnikovKernel) {
+    // bandwidth 1
+    mStepKern[param::VoxX] = static_cast<int>(nearbyint(bwX + 0.5));
+    mStepKern[param::VoxF] = static_cast<int>(nearbyint(bwP + 0.5));
+    mStepKern[param::VoxZ] = static_cast<int>(nearbyint(bwZ + 0.5));
+  } else if (mKernelType == param::GaussianKernel) {
+    // look in ~5 sigma
+    mStepKern[param::VoxX] = static_cast<int>(nearbyint(bwX * 5. + 0.5));
+    mStepKern[param::VoxF] = static_cast<int>(nearbyint(bwP * 5. + 0.5));
+    mStepKern[param::VoxZ] = static_cast<int>(nearbyint(bwZ * 5. + 0.5));
+  } else {
+    LOG(error) << "kernel type " << type << " is not defined";
+  }
+  for (int i = param::VoxDim; i--;) {
+    if (mStepKern[i] < 1) {
+      mStepKern[i] = 1;
+    }
+  }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+///
+/// processing functions
+///
+///////////////////////////////////////////////////////////////////////////////
+
+//______________________________________________________________________________
+void TrackResiduals::processResiduals()
+{
+  if (!mIsInitialized) {
+    init();
+  }
+  for (int iSec = 0; iSec < param::NSectors2; ++iSec) {
+    processSectorResiduals(iSec);
+  }
+}
+
+//______________________________________________________________________________
+void TrackResiduals::processSectorResiduals(int iSec)
+{
+  if (iSec < 0 || iSec > 35) {
+    LOG(error) << "wrong sector: " << iSec;
+    return;
+  }
+  LOG(info) << "processing sector residuals for sector " << iSec;
+  if (!mIsInitialized) {
+    init();
+  }
+  // open file and retrieve data tree (only local files are supported at the moment)
+  std::string filename = "data/tmpDeltaSect";
```

Comment:
please avoid hardcoded names, make it data member of the class with proper default. 
btw, the creation of the name can be done in one step, e.g. 
````
std::string mLocalResFileName = "data/tmpDeltaSect"; // data member
...

std::string filename = mLocalResFileName + std::to_string(iSec) + ".root";
````